### ISSUE TITLE
Soften the git requirement for unversioned usage

### ISSUE THREAD

**Issue by mwpowellhtx** [BrettMayson/HEMTT#308]
**HEMTT Version:** `0.7.4`+
**Project:** private bitbucket repo

**Description:**

Guessing that the hard git requirement should be softened a bit.

**Steps to reproduce:**

`hemtt.toml`:

```TOML
name = "HEQ"
prefix = "heq"
author = "HEQ-Team"
files = [
    "mod.cpp"
    , "meta.cpp"
    , "logo_heq_01.paa"
    , "README.md"
    , "CONTRIBUTORS"
    , "LICENSE"
]

keyname = "{{prefix}}_{{version}}"
# signame = "{{prefix}}_{{version}}-{{git \"id 8\"}}"
signame = "{{prefix}}_{{version}}"

releasebuild = [
    "@zip heq_{{semver.major}}.{{semver.minor}}.{{semver.patch}}"
]

[header_exts]
# version = "{{git \"id 8\"}}"
version = "beta001"
```

`build.bat`:
```BATCH
@echo off
setlocal
set RUST_BACKTRACE=full
rem set RUST_BACKTRACE=1
set hemtt_exe=tools\hemtt\windows\x64\hemtt.exe
if "%1"=="-p" set should_pause=1
%hemtt_exe% build
if "%should_pause%"=="1" pause
endlocal
```

**Additional information:**

Just standing up a "simple" addon, based on ACE, ZEN, etc, references, for the time being. Eventually, however, using the template approach, I suspect, being able to add addons seamlessly, etc.

**HEMTT Output:**

```
$ ./build.bat
HEMTT 0.8.0-a837c9561a871507063208648d8f5304e42e03d1-a837c95
Environment: dev

warning: deprecated value `keyname` in `hemtt.toml` - use `key_name`
warning: deprecated value `signame` in `hemtt.toml` - use `authority`
Clean
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', C:\Users\appveyor\.cargo\git\checkouts\indicatif-741c5cddce652790\250b582\src\progress.rs:106:20
stack backtrace:
   0:        0x1401396c9 - git_index_owner
   1:        0x14015992b - git_index_owner
   2:        0x14012fb44 - git_index_owner
   3:        0x14013c2fc - git_index_owner
   4:        0x14013bf4c - git_index_owner
   5:        0x14013cadf - git_index_owner
   6:        0x14013c665 - git_index_owner
   7:        0x140157670 - git_index_owner
   8:        0x1401575bc - git_index_owner
   9:        0x13ffb0f58 - git_index_owner
  10:        0x13fbf4516 - <unknown>
  11:        0x13fbf3329 - <unknown>
  12:        0x13fb76bb0 - <unknown>
  13:        0x13fb79a45 - <unknown>
  14:        0x13fb6118b - <unknown>
  15:        0x13fb61c76 - <unknown>
  16:        0x14013c507 - git_index_owner
  17:        0x1401499b2 - git_index_owner
  18:        0x14013ce12 - git_index_owner
  19:        0x13fb61267 - <unknown>
  20:        0x140373f69 - git_odb_object_id
  21:         0x7746556d - BaseThreadInitThunk
  22:         0x776c372d - RtlUserThreadStart
```

**Comment by mwpowellhtx** [comment]
I was using the `git bash` prompt in this instance. Discussed in `Discord`, possible issue with progress, did not try with `--ci` or `--no-progress`. Ran instead in a proper Windows command prompt.

```
G:\Source\Arma\Arma 3\Arma 3 Heavy Equipment\working>build.bat
HEMTT 0.8.0-a837c9561a871507063208648d8f5304e42e03d1-a837c95
Environment: dev

warning: .gitignore is missing recommended value `releases/*`
warning: .gitignore is missing recommended value `*.biprivatekey`
warning: .gitignore is missing recommended value `keys/*`
warning: .gitignore is missing recommended value `.hemtt/local`
warning: deprecated value `keyname` in `hemtt.toml` - use `key_name`
warning: deprecated value `signame` in `hemtt.toml` - use `authority`
Clean
Checks         [00:00:00] [##############################] [7|7]
Prebuild       [00:00:01] [##############################] [7|7]
error: Unable to preprocess `script_component.hpp`
    File "\z\ace\addons\main\script_mod.hpp" included from "\\?\G:\Source\Arma\A
rma 3\Arma 3 Heavy Equipment\working\addons\vehicles\script_component.hpp" not f
ound.

error: Unable to build `addons\vehicles`
```

And sorting out those bits.

**Comment by BrettMayson** [comment]
As of 1.0.0 you can easily disable checking for a git hash with
```toml
[version]
git_hash = 0
```

Closed with the release of 1.0.0. If you feel this still needs attention for 1.0.0, please re-open.

**Comment by mwpowellhtx** [comment]
Will let you know as I have an opportunity to circle around on it. Thank you...
